### ISSUE TITLE
allow temporary sharing link for ISA tree

### ISSUE THREAD

**Issue by wittigue** [seek4science/seek#758]
**Describe the bug**
Currently only for single assets the creation of a temporary sharing link is allowed.

**Expected behavior**
Especially for reviewers of manuscripts and their corresponding supplementary data in FAIRDOMHub a temporary sharing link for the ISA tree would be necessary without making the data already public.

**Comment by wittigue** [comment]
now again several users complain about the problem of creating lots of temporary links for reviewers to have one ISA tree as supplementary data
--> could we push it?

**Comment by OlgaKrebs** [comment]
not sure, but would snapshot solve this problem?

**Comment by wittigue** [comment]
@OlgaKrebs snapshot is only possible if all assets are public which users don't like to do during the paper reviewing process

**Comment by OlgaKrebs** [comment]
I know, but I suggest that we can may be change it (why not?)  - allow snapshots for "internal use" before publishing/getting DOI. cause procedure is = 1. create snapshot 2. get DOI 
I would rather go this way (temporary link for snapshot) than allow temporary link for whole ISA tree as error prawn

**Comment by floradanna** [comment]
I think the procedure currently is: 1. Make items public in SEEK, 2. Make a snapshot, 3. get DOI.

The problematic step for researchers seems to be the number 1 (making items public in seek), if I understand correctly what Ulrike is saying.

**Comment by stuzart** [comment]
https://docs.seek4science.org/images/user-guide/snapshotting_2.png

**Comment by OlgaKrebs** [comment]
BTW: just checked, I can create snap shot for my Investigation containing non-public elements https://fairdomhub.org/investigations/1/  , it means I went through the whole procedure to the end .... but no new snapshot created (just 2 olds) . Bug? will check it and report extra

**Comment by floradanna** [comment]
@stuzart Yes, but if items are not included in the snapshot, they cannot be shared/made accessible to reviewers via that snapshot, correct?

Researchers are asking for a way to **share an ISA-tree of private items** with non-registered seek users via 1 link for the entire ISA, without having to make those items public in seek (if I understand correctly what Ulrike is asking).

**Comment by whomingbird** [comment]
ISA Temporary Links allow you to share private Investigations, Studies, and Assays with external collaborators or reviewers by creating time-limited access links. These links enable viewing and downloading without requiring user accounts in SEEK, 

**Features**
Hierarchical Access

- Investigation Link: Grants access to the Investigation and ALL Studies, Assays, and associated assets within it
- Study Link: Grants access to the Study and ALL Assays and associated assets within it
- Assay Link: Grants access to the Assay and ALL associated assets (DataFiles, Models, SOPs, Documents, etc.)

**Key Characteristics**

- Downward propagation: Parent codes grant access to all children
- No upward propagation: Child codes do NOT grant access to parents
- Time-limited: Each link has an expiration date
- Secure: Uses cryptographically secure random codes
- Works with private children: Public items can share codes to grant access to private children

**Example Link:**

`https://your-seek-instance.org/investigations/123?code=ABC123XYZ789...`

also possible accessing with Code via API
`GET /investigations/123.json?code=GENERATED_CODE_HERE
`

**Comment by whomingbird** [comment]
https://github.com/seek4science/seek-documentation/blob/408c00e2b62c24ca91434bc0cfb29194f167d1af/help/user-guide/isa_temporary_links_user_guide.md
